### PR TITLE
Fix flaky templatized call

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Fixes flaky test (example in #20795) https://github.com/apache/airflow/runs/4793994282?check_suite_focus=true

File: tests/sensors/test_python.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def test_python_callable_arguments_are_templatized(self):
 
         ds_templated = DEFAULT_DATE.date().isoformat()
         # 2 calls: first: at start, second: before timeout
-        assert 2 == len(recorded_calls)
+        assert 1 <= len(recorded_calls)
```
Comment (uranusjr) [inline tests/sensors/test_python.py]: This seems to defeat the purpose of this assertion. Why is the second call not recorded?

Comment (uranusjr) [reply tests/sensors/test_python.py]: Why do we have the second call at all? Can we not have it?

Comment (potiuk) [reply tests/sensors/test_python.py]: It' sensor and number of calls depends on how fast the whole system is at all (sometimes we will not get the second one if the system is very busy with other parallell tests. 

We know tht the first call happens for sure (it is immediate), and most of the time the second too. The assert is actually correct IMHO ("expect at least 1 call").  We only check the first call anyway (and this is the gist of it as we really want to see if JINJA template works in the sensor call). 

If you think on how to fix it better I am all ears :)

Comment (potiuk) [reply tests/sensors/test_python.py]: The thing here is that his is the "timing" tests  and we test it with "run" method and expect timeout. 

In perfect world, we could likely use freezegun and simulate the exact time passing, but IMHO it is totally not worth it if we can 100% fix it with this three characters change. Trading test simplicity with perfectness.

Comment (uranusjr) [reply tests/sensors/test_python.py]: Let’s remove this assert entirely then. The `1 <=` part is validated below anyway (if there isn’t a call, `recorded_calls[0]` would automatically fail).

Comment (potiuk) [reply tests/sensors/test_python.py]: Works for me :)

Comment (potiuk) [reply tests/sensors/test_python.py]: I also removed it from the test below (same story but it had a comment about it and was already using >= 1).

**Review by uranusjr — APPROVED** [review]

**Review by ashb — APPROVED** [review]

**Pull request merged**